iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveTableUtil.java

```diff
@@ -168,6 +174,31 @@ private static List<DataFile> getDataFiles(RemoteIterator<LocatedFileStatus> fil
     return dataFiles;
   }
 
+  public static void appendFiles(URI fromURI, String format, Table icebergTbl, boolean isOverwrite, Configuration conf)
+      throws SemanticException {
+    try {
+      Transaction transaction = icebergTbl.newTransaction();
+      if (isOverwrite) {
+        DeleteFiles delete = transaction.newDelete();
+        delete.deleteFromRowFilter(Expressions.alwaysTrue());
+        delete.commit();
+      }
+      AppendFiles append = transaction.newAppend();
+      PartitionSpec spec = icebergTbl.spec();
+      MetricsConfig metricsConfig = MetricsConfig.fromProperties(icebergTbl.properties());
+      String nameMappingString = icebergTbl.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
```

Comment:
what's gonna happen if nameMappingString is null?

Reply:
it is used for metrics, the ``null`` case is handled 
```
      return nameMapping != null ? ParquetSchemaUtil.applyNameMapping(type, nameMapping) : ParquetSchemaUtil.addFallbackIds(type);
```